sqlmesh/core/model/definition.py

```diff
@@ -1813,17 +1821,25 @@ def _create_model(
     dialect = dialect or ""
     physical_schema_override = physical_schema_override or {}
 
+    # Assumption: any specified default that is not a ModelMeta field is kind-specific.
```

Comment:
Seems like a dangerous assumption. Should we be more specific in our defaults and eg. have defaults per model kind?

Reply:
I originally had a private field in the model defaults that listed which fields were kind-specific, peeled those off here, then matched them against all of a kind's fields in the kind validator. 

Downside of that approach is that if we expose a kind-specific default for any kind it would be set for all kinds that have that field. 

Doing it per kind seems like a better choice - it would set us up to be more granular in the future, such that we could expose defaults for only a subset of kinds.